src/axom/core/Map.hpp

```diff
@@ -0,0 +1,755 @@
+// Copyright (c) 2017-2021, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#ifndef AXOM_MAP_HPP_
+#define AXOM_MAP_HPP_
+
+#include "axom/config.hpp"                  // for compile-time defines
+#include "axom/core/Macros.hpp"             // for axom macros
+#include "axom/core/memory_management.hpp"  // for memory allocation functions
+#include "axom/core/Types.hpp"              // for axom types
+
+// C/C++ includes
+#include <functional>  //for hashing until a custom method is defined
+#include <iostream>
+
+namespace axom
+{
+namespace experimental
+{
+namespace axom_map
```

Comment:
was this namespace discussed? `axom::experimental::axom_map` is long and repetitive

Reply:
https://github.com/LLNL/axom/pull/589#discussion_r653066151

Reply:
The axom_map part is also internal -- the Map itself resides at axom::experimental::map as its own class.